exercise2/Colors.java

```diff
@@ -1,16 +1,30 @@
-package exercise2;
+import java.util.*;
+import java.util.stream.*;
 
 public class Colors {
+    String[] colorful;
+    public Colors (String[] strings) {
 
-    String[] colorful(String[] strings) {
-        // For the given strings, keep all the strings that begin with a color ("red", "yellow", "green", "blue", etc.) return an array of strings that start with a color.  discard all the other strings.  If no strings start with a color in the input array, return an empty array.
-        return new String[]{};
+        // For the given strings, keep all the strings that begin with a color ("red", "yellow", "green", "blue", etc.)
+        // return an array of strings that start with a color.  discard all the other strings.
+        // If no strings start with a color in the input array, return an empty array.
+
+        this.colorful = strings;
     }
 
     public static void main(String[] args) {
-        Colors code = new Colors();
-        code.colorful(new String[]{"bluebird","red herring", "yellow bumblebee","black spider", "green monkey","pink","great gatsby","feeding frenzy","black bird","widowmaker","quran","bible","torah"});
+
+        String[] strings = {"bluebird","red herring", "yellow bumblebee","black spider", "green monkey","pink","great gatsby","feeding frenzy","black bird","widowmaker","quran","bible","torah"};
+        Colors code = new Colors(strings);
         System.out.println("This is the Colors class");
+        String[] r =
```

Comment:
it would be nice to make a (static?) array to compare against, so you could add new values later without having to dig into the code